### PR TITLE
Fix Meilisearch error and refactor of the server index.js

### PR DESCRIPTION
**Pull request by berry-13**
## Summary

1. Fix Meilisearch & remove useless Meilisearch console log
2. Small index.js refactor

## Change Type

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

Check if there's still the console log & if all the social login work (because of a small refactor)

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented in any complex areas of my code
- [x] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

File: api/server/index.js [hunk 1]
```diff
@@ -97,21 +68,55 @@ config.validate(); // Validate the config
...
+        `Server listening on all interfaces at port ${port}. Use http://localhost:${port} to access it`,
       );
     } else {
       console.log(`Server listening at http://${host == '0.0.0.0' ? 'localhost' : host}:${port}`);
     }
   });
-})();
+};
+
+const configureSocialLogins = (app) => {
```
Comment (danny-avila) [inline api/server/index.js]: if this is being factored out as a function, it can go in a separate file.

File: api/lib/db/indexSync.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ let currentTimeout = null;
 
 // eslint-disable-next-line no-unused-vars
 async function indexSync(req, res, next) {
+  if (process.env.SEARCH && process.env.SEARCH.toLowerCase() !== 'true') {
```
Comment (danny-avila) [inline api/lib/db/indexSync.js]: this seems redundant with line 12, try testing this and if it matches the expected change
```js
const searchEnabled = process.env?.SEARCH?.toLowerCase() === 'true';

if (!searchEnabled) {
  return;
}
```

**Review by danny-avila — CHANGES_REQUESTED** [review]

File: api/server/index.js (hunk 1 above)
Comment (berry-13) [reply api/server/index.js]: I have tested with Google, Github, and Discord, and the login works.

**Comment by Berry-13** [comment]
Do you mean to create a new file like this @danny-avila?

**Comment by danny-avila** [comment]
> Do you mean to create a new file like this @danny-avila?

yes thank you, that will help de-clutter the main index file, is the PR ready?

**Comment by Berry-13** [comment]
Yes, I have tried to test most of the logins and everything seems to work fine, even the tests run successfully

**Review by danny-avila — APPROVED** [review]

**Pull request merged**